modules/iam/main.tf

```diff
@@ -0,0 +1,234 @@
+        "ec2:ModifyNetworkInterfaceAttribute",
+        "ec2:ReleaseAddress",
+        "autoscaling:CompleteLifecycleAction",
+        "autoscaling:DescribeAutoScalingGroups",
+        "elasticloadbalancing:RegisterTargets",
+        "elasticloadbalancing:DeregisterTargets"
+      ]
+
+      resources = ["*"]
+
+      condition = {
+        test     = "StringEquals"
+        variable = "aws:ResourceTag/Owner"
+        values   = [var.global_tags["Owner"]]
+      }
+    }
+    statement3 = {
+      sid    = "3"
+      effect = "Allow"
+      actions = [
+        "ec2:DescribeAddresses",
+        "ec2:DescribeInstances",
+        "ec2:DescribeNetworkInterfaces",
+        "ec2:DescribeTags",
+        "ec2:DescribeSubnets",
+      ]
+
+      resources = ["*"]
+    }
+    statement4 = {
+      sid    = "4"
+      effect = "Allow"
+      actions = [
+        "kms:GenerateDataKey*",
+        "kms:Decrypt",
+        "kms:CreateGrant"
+      ]
+
+      resources = ["*"]
+
+      condition = {
+        test     = "StringEquals"
+        variable = "aws:ResourceTag/Owner"
+        values   = [var.global_tags["Owner"]]
+      }
+    }
+  }
+
+  lambda_delicense_policy = {
+    statement1 = {
+      sid    = "1"
+      effect = "Allow"
+      actions = [
+        "ssm:DescribeParameters",
+        "ssm:GetParametersByPath",
+        "ssm:GetParameter",
+        "ssm:GetParameterHistory"
+      ]
+
+      resources = [
+        "arn:${data.aws_partition.this.partition}:ssm:${var.region}:${local.account_id}:parameter${local.delicense_param}"
+      ]
+    }
+  }
+
+  vmseries_policy = {
+    statement1 = {
+      sid    = "1"
+      effect = "Allow"
+      actions = [
+        "cloudwatch:PutMetricData",
+        "cloudwatch:GetMetricData",
+        "cloudwatch:ListMetrics"
+      ]
+
+      resources = [
+        "*"
+      ]
+    }
+    statement2 = {
+      sid    = "2"
+      effect = "Allow"
+      actions = [
+        "cloudwatch:PutMetricAlarm",
+        "cloudwatch:DescribeAlarms"
+      ]
+
+      resources = [
+        "arn:${data.aws_partition.this.partition}:cloudwatch:${var.region}:${data.aws_caller_identity.this.account_id}:alarm:*"
+      ]
+    }
+  }
+
+  bootstrap_policy = {
+    statement1 = {
+      sid       = "1"
+      effect    = "Allow"
+      actions   = ["s3:GetObject"]
+      resources = ["arn:${data.aws_partition.this.partition}:s3:::${var.aws_s3_bucket}"]
+    }
+    statement2 = {
+      sid       = "2"
+      effect    = "Allow"
+      actions   = ["s3:GetObject"]
+      resources = ["arn:${data.aws_partition.this.partition}:s3:::${var.aws_s3_bucket}/*"]
+    }
+  }
+
+  aws_policies = {
+    "custom" = {
+      enable     = var.custom_policy == null ? false : true
+      definition = try(var.custom_policy, null)
+    },
+    "lambda_execute" = {
+      enable     = var.create_lambda_policy && var.delicense_ssm_param_name != null ? true : false
+      definition = try(local.lambda_execute_policy, null)
```

Comment:
Except for the custom, all other policies are always non empty, so I think using `try(...)` for them is excessive.

Reply:
yes, they are predefined